lnst/Recipes/ENRT/SRIOVNetnsRecipe.py

```diff
@@ -0,0 +1,158 @@
+
+    .. code-block:: none
+
+                    +--------+
+             +------+ switch +-----+
+             |      +--------+     |
+          +--+-+                 +-+--+
+    +-----|eth0|-----+     +-----|eth0|-----+
+    |     +----+     |     |     +----+     |
+    |       |        |     |       |        |
+    |  +-namespace-+ |     |  +-namespace-+ |
+    | |    vf0     | |     | |    vf0     | |
+    | +-----------+  |     | +-----------+  |
+    |     host1      |     |     host2      |
+    +----------------+     +----------------+
+
+    All sub configurations are included via Mixin classes.
+
+    The actual test machinery is implemented in the :any:`BaseEnrtRecipe` class.
+    """
+    host1 = HostReq()
+    host1.eth0 = DeviceReq(label="net1", driver=RecipeParam("driver"))
+
+    host2 = HostReq()
+    host2.eth0 = DeviceReq(label="net1", driver=RecipeParam("driver"))
+
+    offload_combinations = Param(default=(
+        dict(gro="on", gso="on", tso="on", tx="on", rx="on"),
+        dict(gro="off", gso="on", tso="on", tx="on", rx="on"),
+        dict(gro="on", gso="off", tso="off", tx="on", rx="on"),
+        dict(gro="on", gso="on", tso="off", tx="off", rx="on"),
+        dict(gro="on", gso="on", tso="on", tx="on", rx="off")))
+
+    def test_wide_configuration(self):
+        """
+        Test wide configuration for this recipe involves just adding an IPv4 and
+        IPv6 address to the matched eth0 nics on both hosts.
+
+        host1.eth0 = 192.168.101.1/24 and fc00::1/64
+
+        host2.eth0 = 192.168.101.2/24 and fc00::2/64
+        """
+        host1, host2 = self.matched.host1, self.matched.host2
+        configuration = super().test_wide_configuration()
+        configuration.test_wide_devices = []
+
+        for i, host in enumerate([host1, host2]):
+            host.run(f"devlink dev eswitch set pci/{host.eth0.bus_info} mode switchdev")
+            host.run(f"echo 1 > /sys/class/net/{host.eth0.name}/device/sriov_numvfs")
+
+            vf_ifname = dict(ifname=f"{host.eth0.name}v0")
+            host.map_device("vf_eth0", vf_ifname)
+
+            host.newns = NetNamespace(f"lnst{i+1}")
+            host.newns.vf_eth0 = host.vf_eth0
+
+            host.newns.vf_eth0.ip_add(ipaddress("192.168.101." + str(i+1) + "/24"))
+            host.newns.vf_eth0.ip_add(ipaddress("fc00::" + str(i+1) + "/64"))
+
+            host.eth0.up()
+            host.newns.vf_eth0.up()
+
+            configuration.test_wide_devices.append(host.newns.vf_eth0)
+
+        self.wait_tentative_ips(configuration.test_wide_devices)
+
+        return configuration
+
+    def generate_test_wide_description(self, config):
+        """
+        Test wide description is extended with the configured addresses
+        """
+        desc = super().generate_test_wide_description(config)
+        desc += [
+            "Configured {}.{}.ips = {}".format(
+                dev.host.hostid, dev.name, dev.ips
+            )
```

Comment:
you can probably add the namespace name here.

Reply:
Changed whole test wide description, fixed.